### ISSUE TITLE
Assertion error

### ISSUE THREAD

**Issue by Oscar310118** [simonwu53/NetCalib2-Sensors-Auto-Calibration#3]
Hi, I have already generated camera depth data. Here is my code. But after that, I got an assertion error. Is it because of my code, thanks.
![error](https://user-images.githubusercontent.com/66508452/165724870-fbeffb80-cd96-4686-a742-c6e9c126f34e.jpg)

![image](https://user-images.githubusercontent.com/66508452/165725228-8333a8b9-54d0-42fe-8ed6-db3f4a6b58cd.png)

**Comment by simonwu53** [comment]
Yes, please check your input data.

**Comment by Oscar310118** [comment]
Is it possible to provide your cv2.StereoSGBM_create parameter, thanks?

**Comment by simonwu53** [comment]
Here's some example settings. It's upon you to decide whether these are suitable for your project. 

> ### Settings 0
> minDisparity: 0
> numDisparities: 80
> block_size: 5
> disp12MaxDiff: 7
> speckleWindowSize: 100
> speckleRange: 11
> uniquenessRatio: 0
> preFilterCap: 7
> P1: 170
> P2: 1678
> Lambda: 8000
> SigmaColor: 20
> 
> ### Settings 1
> minDisparity: 0
> numDisparities: 160
> block_size: 5
> disp12MaxDiff: 1
> speckleWindowSize: 0
> speckleRange: 2
> uniquenessRatio: 15
> preFilterCap: 63
> P1: 216
> P2: 864
> Lambda: 80000
> SigmaColor: 12
> 
> 
> ### Settings 2
> minDisparity: 0
> numDisparities: 160
> block_size: 5
> disp12MaxDiff: 1
> speckleWindowSize: 180
> speckleRange: 2
> uniquenessRatio: 15
> preFilterCap: 63
> P1: 18
> P2: 2000
> Lambda: 8000
> SigmaColor: 12

**Comment by Oscar310118** [comment]
Hi, I try to use your setting1, but it seems that cv2.StereoSGBM_create doesn't have Lambda and SigmaColor parameters. Is it necessary to change them, thanks?

**Comment by Oscar310118** [comment]
After using your setting, it still failed. I tried to show np.max(depth_ong), but it seems that if the np.max(depth_ong) is a floating number, it will fail. Here is my result.
![image](https://user-images.githubusercontent.com/66508452/166190175-afd5675b-2b9b-4847-b92d-0c9cc495ced0.png)

**Comment by Oscar310118** [comment]
Can you provide your method to save the depth map? I use cv2.imwrite but something wrong with it, thanks.

**Comment by simonwu53** [comment]
Why not first check the source code? In "preprocess.py" I have written those lines how to save a depth map.

**Comment by Oscar310118** [comment]
Do you mean code like below? I'm confused that how to save my SGBM result to the depth map. It seems that the range of the depth map is [0,1].
 
![image](https://user-images.githubusercontent.com/66508452/166627595-a6a9ed75-2cb6-4924-8c7f-bdb7818d2930.png)

**Comment by simonwu53** [comment]
No, it’s in the file. You can find it if you read it thoroughly.

**Comment by Oscar310118** [comment]
I found cv2.imwrite in the code. 
![image](https://user-images.githubusercontent.com/66508452/166865018-9460ea3e-8300-4731-bd23-558be28a4802.png)
Where is the part I can integrate my SGBM algorithm in preprocess.py? Here is my SGBM code.
![image](https://user-images.githubusercontent.com/66508452/166865271-04237be1-35f8-4623-bbac-f45c2959af8b.png)

**Comment by simonwu53** [comment]
As I said before, this is the part you need to write your own. I’m not responsible for your project, coding or debugging issues. It’s your job. So, you’ve found the code to save depth maps, you should have the ability to understand what kind of data to save in terms of its dtype, value ranges.

By the way, pay attention to the depth maps’ sizes for camera and LiDAR. I’ve updated README for the requirements of depth maps’ sizes, which could be found only in the code before. In general, depth maps’ sizes from the camera and LiDAR should be the same. And this is the cause of your [original error](https://github.com/simonwu53/NetCalib2-Sensors-Auto-Calibration/issues/3#issue-1218471481).
